### PR TITLE
support multiple ego vehicles

### PR DESCRIPTION
**Pull request by makaveli10**
add support for publishing bboxes + lidar_overlay when multiple ego vehicles are spawned

### CODE REVIEW

File: carla_ros_bridge/src/carla_ros_bridge/bridge.py [hunk 1]
```diff
@@ -297,55 +307,60 @@ def publish_bboxes_and_lidar_overlay(self):
...
-            lidar_data = self.lidar.get_lidar_data()
-            if self.lidar_to_camera is not None:
-                if lidar_data is not None:
-                    if lidar_data.frame == frame:
-                        # draw
-                        image_w = float(rgb_cam.carla_actor.attributes.get("image_size_x"))
-                        image_h = float(rgb_cam.carla_actor.attributes.get("image_size_y"))
-                        img = self.lidar_to_camera.lidar_overlay(lidar_data, img, rgb_cam, image_w, image_h)
+        for parent, rgb_cams in self.rgb_cams.items():
+            # check if lidar for the smae parent exists
```
Comment (zoq) [inline carla_ros_bridge/src/carla_ros_bridge/bridge.py]:
```suggestion
            # check if lidar for the same parent exists
```

**Comment by makaveli10** [comment]
@zoq I figured that ```try``` block was necessary there incase the lidar doesnt exist for that parent(ego vehicle)

File: carla_ros_bridge/src/carla_ros_bridge/bridge.py [hunk 2]
```diff
@@ -297,55 +307,66 @@ def publish_bboxes_and_lidar_overlay(self):
...
-                    if lidar_data.frame == frame:
-                        # draw
-                        image_w = float(rgb_cam.carla_actor.attributes.get("image_size_x"))
-                        image_h = float(rgb_cam.carla_actor.attributes.get("image_size_y"))
-                        img = self.lidar_to_camera.lidar_overlay(lidar_data, img, rgb_cam, image_w, image_h)
+        for parent, rgb_cams in self.rgb_cams.items():
+            # check if lidar exists for the same parent
+            curr_lidar = None
+            try:
+                curr_lidar = self.lidar[parent]
```
Comment (zoq) [inline carla_ros_bridge/src/carla_ros_bridge/bridge.py:314]: Would be cleaner if you add a check instead:

```python
if parent in self.lidar:
  curr_lidar = self.lidar[parent]
else:
  self.node.logwarn(f"lidar is None for {parent}")
```

Comment (makaveli10) [reply carla_ros_bridge/src/carla_ros_bridge/bridge.py:314]: yep right. thanks!

**Review by zoq — APPROVED** [review]
No further comments from my side, looks great.

**Pull request merged**